### PR TITLE
Support basic auto-completion

### PR DESCRIPTION
**Pull request by psiinon**

### CODE REVIEW

File: src/org/zaproxy/zap/extension/scripts/ConsolePanel.java [hunk 1]
```diff
@@ -47,18 +47,22 @@
...
 import org.zaproxy.zap.utils.FontUtils;
 import org.zaproxy.zap.view.LayoutHelper;
+import org.zaproxy.zap.view.ZapToggleButton;
 
 public class ConsolePanel extends AbstractPanel implements Tab {
 
 	private static final long serialVersionUID = 1L;
 
 	private static final String BASE_NAME_SCRIPT_EXECUTOR_THREAD = "ZAP-ScriptExecutor-";
+    private static final ImageIcon AUTO_COMPLETE_ICON = new ImageIcon(OutputPanel.class.getResource(
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/scripts/ConsolePanel.java]: Could be aligned with above statement.

File: src/org/zaproxy/zap/extension/scripts/ConsolePanel.java [hunk 2]
```diff
@@ -114,7 +118,8 @@ private void initialize() {
 			
 			panelToolbar.add(this.getRunButton(), LayoutHelper.getGBC(0, 0, 1, 0.0D));
 			panelToolbar.add(this.getStopButton(), LayoutHelper.getGBC(1, 0, 1, 0.0D));
-			panelToolbar.add(this.getScriptTitle(), LayoutHelper.getGBC(2, 0, 1, 0.0D));
+            panelToolbar.add(this.getAutoCompleteButton(), LayoutHelper.getGBC(2, 0, 1, 0.0D));
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/scripts/ConsolePanel.java]: Ditto.

File: src/org/zaproxy/zap/extension/scripts/ConsolePanel.java [hunk 3]
```diff
@@ -185,6 +190,25 @@ public void actionPerformed(java.awt.event.ActionEvent e) {
 		return stopButton;
 	}
 
+    private ZapToggleButton getAutoCompleteButton() {
+        if (autoCompleteButton == null) {
+            autoCompleteButton = new ZapToggleButton();
+            autoCompleteButton.setIcon(DisplayUtils.getScaledIcon(AUTO_COMPLETE_ICON));
+            autoCompleteButton.setSelectedIcon(DisplayUtils.getScaledIcon(AUTO_COMPLETE_ICON));
+            autoCompleteButton.setToolTipText(Constant.messages.getString("scripts.toolbar.tooltip.autocomplete.disabled"));
+            autoCompleteButton.setSelectedToolTipText(Constant.messages.getString("scripts.toolbar.tooltip.autocomplete.enabled"));
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/scripts/ConsolePanel.java]: Should the tool tips be the other way around? (It should indicate what will happen if clicked, not the current state?)

Comment (psiinon) [reply src/org/zaproxy/zap/extension/scripts/ConsolePanel.java]: Was going to have them like that, but saw the script and scroll lock buttons show the current state so copied them for consistency...

**Comment by psiinon** [comment]
Have fixed the other issues (inc an NPE). 
Keep the tooltips or change the existing ones as well?

**Comment by psiinon** [comment]
Following chat on irc have updated all the buttons to reflect what happens when they are pushed.

**Comment by thc202** [comment]
Thanks!

File: src/org/zaproxy/zap/extension/scripts/resources/Messages.properties [hunk 4]
```diff
@@ -5,12 +5,12 @@
...
-scripts.output.clearOnRun.button.disabled.toolTip	= Clear script output panel on run: disabled
-scripts.output.clearOnRun.button.enabled.toolTip	= Clear script output panel on run: enabled
-scripts.output.scriptLock.button.enabled.toolTip	= Script lock: enabled - only output from the selected script will be displayed
-scripts.output.scriptLock.button.disabled.toolTip	= Script lock: disabled - output from all scripts will be displayed
-scripts.output.scrolllock.button.disabled.toolTip	= Scroll lock: disabled
-scripts.output.scrolllock.button.enabled.toolTip	= Scroll lock: enabled
+scripts.output.clearOnRun.button.disabled.toolTip	= Enable clear script output panel on run
+scripts.output.clearOnRun.button.enabled.toolTip	= Disable clear script output panel on run
+scripts.output.scriptLock.button.enabled.toolTip	= Disable script lock - output from all scripts will then be displayed
+scripts.output.scriptLock.button.disabled.toolTip	= Enable Script lock - only output from the selected script will then be displayed
```
Comment (kingthorin) [inline src/org/zaproxy/zap/extension/scripts/resources/Messages.properties]: Remove caps on script?

**Comment by psiinon** [comment]
Fixed :)

**Pull request merged**